lib/fs.js

```diff
@@ -1160,7 +1160,7 @@ function FSWatcher() {
   this._handle.onchange = function(status, event, filename) {
     if (status < 0) {
       self._handle.close();
-      self.emit('error', errnoException(status, 'watch'));
+      self.emit('error', errnoException(status, 'watch ' + filename));
```

Comment:
Instead of tacking the filename on here, it might be more useful to create the error object, then attach the filename to the error object.